src/logical/execution/planning/negation.rs

```diff
@@ -0,0 +1,142 @@
+//! Functionality for creating an execution plan that implements negation.
+
+use std::collections::HashMap;
+
+use crate::{
+    logical::{
+        execution::planning::plan_util::{compute_filters, subplan_union_reordered},
+        model::{Atom, Filter, Term, Variable},
+        program_analysis::variable_order::VariableOrder,
+        table_manager::SubtableExecutionPlan,
+        types::LogicalTypeEnum,
+        TableManager,
+    },
+    physical::{
+        management::{database::ColumnOrder, execution_plan::ExecutionNodeRef},
+        tabular::operations::{triescan_minus::SubtractInfo, triescan_project::ProjectReordering},
+        util::mapping::traits::NatMapping,
+    },
+};
+
+struct AtomInfo {
+    restricted_variable_order: VariableOrder,
+    reorder: ColumnOrder,
+    projection: ProjectReordering,
+    subtraction: SubtractInfo,
```

Comment:
More of a general comment:
I think also in other places many helper structs are named in a way that does not convey much information of what is stored in them. For instance, `AtomInfo` could basically be anything. On the other hand `SubtractInfo` at least gives a hint on what this data is used for but the contents are also not clear from the name.
At least for me that makes understanding what is supposed to happen in the code quite hard.

Reply:
Well the thing is that those really are just "random" collections of objects that just need to be used together. As such, the overall struct does not represent any kind of concept that can have a meaningful name. 

I changed `AtomInfo` to `AtomNegationInfo` and added a bit more comments. But I don't think its going to get better than that.

Reply:
Once we have a separate rule representation from which execution plans can be built more easily, much of this code will change anyways, so I wouldn't worry about it too much

Reply:
Sounds good :)